tests/test_inference_api_eval_wrapper.py

```diff
@@ -0,0 +1,131 @@
+                    'num_fewshot': [0, 1],
+                    'icl_task_type':
+                        'language_modeling',
+                    'continuation_delimiter':
+                        '\nAnswer: ',
+                    'has_categories':
+                        True
+                })
+            ])
+    })
+
+
+def mock_create(**kwargs: Dict[str, str]):
+    prompt = kwargs['prompt']
+    if prompt == 'AMERICAN HISTORY: On May 29, 1765 Patrick Henrys Stamp Act protest was interrupted with this one word\nAnswer:':  # pyright: ignore[reportUnnecessaryComparison]
+        return {
+            'choices': [{
+                'logprobs': {
+                    'top_logprobs': [{
+                        ' Tre': 0,
+                    }],
+                },
+            }],
+        }
+    elif prompt == 'AMERICAN HISTORY: On May 29, 1765 Patrick Henrys Stamp Act protest was interrupted with this one word\nAnswer: Tre':  # pyright: ignore[reportUnnecessaryComparison]
+        return {
+            'choices': [{
+                'logprobs': {
+                    'top_logprobs': [{
+                        'ason': 0,
+                    }],
+                },
+            }],
+        }
+    elif prompt == 'AMERICAN HISTORY: On May 29, 1765 Patrick Henrys Stamp Act protest was interrupted with this one word\nAnswer: Treason':  # pyright: ignore[reportUnnecessaryComparison]
+        return {
+            'choices': [{
+                'logprobs': {
+                    'top_logprobs': [{
+                        '!': 0,
+                    }],
+                },
+            }],
+        }
+    else:
+        raise ValueError(
+            f'Mocked `openai.Completion.create` received invalid prompt: {prompt}.'
+        )
+
+
+def test_openai_api_eval_wrapper(tmp_path: str):
+    with patch('openai.Completion') as mock:
+        mock.create = mock_create
+        model_name = 'davinci'
+        tokenizer = OpenAITokenizerWrapper(model_name)
+        model = OpenAICausalLMEvalWrapper(model_cfg={'version': model_name},
+                                          tokenizer=tokenizer)
+        task_cfg = load_icl_config()
+        evaluators, _ = build_icl_evaluators(task_cfg.icl_tasks,
+                                             tokenizer,
+                                             1024,
+                                             1,
+                                             destination_dir=str(tmp_path))
+
+        batch = next(evaluators[0].dataloader.dataloader.__iter__())
+        result = model.eval_forward(batch)
+        model.update_metric(batch,
+                            result,
+                            metric=model.get_metrics()
+                            ['InContextLearningLMAccuracy'])  # pyright: ignore
+        acc = model.get_metrics(
+        )['InContextLearningLMAccuracy'].compute(  # pyright: ignore
+        )  # pyright: ignore
+        assert acc == 1.0
```

Comment:
can you construct the example such that the accuracy is not 1? just a better test if its something between 0 and 1